M2/Macaulay2/e/aring-gf-flint-big.hpp

```diff
@@ -11,6 +11,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wconversion"
 #include <flint/flint.h>       // for flint_free, flint_rand_t, fmpz_t
+#include <gmp.h>               // required for FLINT functions that use GMP
```

Comment:
I don't think we include `gmp.h` directly, instead you should probably include `math-include.h`.

Reply:
We do in a few places:

```m2
$ grep -r "<gmp.h>"
NCAlgebras/FreeAlgebra.hpp:#include <gmp.h>                      // for mpz_srcptr, mpq_srcptr
NCAlgebras/FreeAlgebraQuotient.hpp:#include <gmp.h>                      // for mpz_srcptr, mpq_srcptr
schreyer-resolution/res-f4-m2-interface.cpp:#include <gmp.h>                                          // for mpz_clear
```

I don't see the advantage of including another header that includes even more headers we won't be using.

Reply:
Not really: the first two include math-include.h via ringelem.hpp, the third includes it via gbring.hpp, so in all cases including gmp.h is redundant and should be removed.

The point is to make sure every time certain libraries are loaded, the same macros.

Reply:
I've updated that commit with your suggestion, and also removed the redundant gmp.h includes from those other files.